### PR TITLE
Fragment basic operations

### PR DESCRIPTION
**Pull request by Arthoni**
* GitUIProxy operation
* DevStudio action

(closes OpenFLUID/openfluid#1111)

### CODE REVIEW

File: src/openfluid/ui/waresdev/FragmentsImportWorker.cpp [hunk 1]
```diff
@@ -145,28 +145,25 @@ bool FragmentsImportWorker::importElement(const QString& GitUrl, const QString&
...
       // removing also the reference in .git folder
-      openfluid::tools::FilesystemPath GitSubmoduleTargetDirectory = openfluid::tools::FilesystemPath(
-        {RootPath.toStdString(), ".git", "modules", DestSubPath.toGeneric()});
-      if (GitSubmoduleTargetDirectory.isDirectory())
+      if (openfluid::ui::waresdev::cleanModuleDir(RootPath, DestSubPath) == 0)
       {
-        GitSubmoduleTargetDirectory.removeDirectory();
         emit info(tr("Fragment directory removed from .git/modules since import not successful"));
       }
+
```
Comment (jctophefabre) [inline src/openfluid/ui/waresdev/FragmentsImportWorker.cpp:152]: I miss some information to clearly understand this part (current and following lines)

File: src/openfluid/ui/waresdev/GitUIProxy.cpp [hunk 2]
```diff
@@ -225,9 +231,91 @@ bool GitUIProxy::addSubmodule(const QString& FromUrl, const QString& ToPath, con
...
+  }
+  return false;
+}
+
+
+// =====================================================================
+// =====================================================================
+
+
+QString GitUIProxy::removeSubmodule(const QString& MainPathString, const QString& SubmodulePathString)
```
Comment (jctophefabre) [inline src/openfluid/ui/waresdev/GitUIProxy.cpp:250]: I need a description of removal process of a git submodule during the code review

File: src/openfluid/ui/waresdev/GitUIProxy.cpp [hunk 3]
```diff
@@ -225,9 +231,91 @@ bool GitUIProxy::addSubmodule(const QString& FromUrl, const QString& ToPath, con
...
+
+
+// =====================================================================
+// =====================================================================
+
+
 bool GitUIProxy::clone(const QString& FromUrl, const QString& ToPath,
-                       const QString& Username, const QString& Password,
-                       bool SslNoVerify, const QString& LocalGitRepoPath, bool WithoutVersioning)
+                     const QString& Username, const QString& Password,
```
Comment (jctophefabre) [inline src/openfluid/ui/waresdev/GitUIProxy.cpp:317]: Indentation seems incorrect 🧐

File: src/openfluid/ui/waresdev/WareSrcExplorer.cpp [hunk 4]
```diff
@@ -432,6 +439,41 @@ void WareSrcExplorer::onNewFragmentAsked()
...
+    );
+
+  if (QMessageBox::warning(this, tr("Delete fragment"), tr("Are you sure you want to delete \"%1\"?").arg(SubPath),
+                           QMessageBox::Ok | QMessageBox::Cancel, QMessageBox::Cancel) == QMessageBox::Cancel)
+  {
+    return;
+  }
+  
+  openfluid::ui::waresdev::WareGitDialog Dialog;
+  Dialog.setWindowTitle("git rm submodule");
```
Comment (jctophefabre) [inline src/openfluid/ui/waresdev/WareSrcExplorer.cpp:467]: Title to modify. This title is too rough and should be translatable

File: src/openfluid/ui/waresdev/WareSrcExplorer.cpp [hunk 5]
```diff
@@ -432,6 +439,41 @@ void WareSrcExplorer::onNewFragmentAsked()
...
+  Dialog.setContent(Git.removeSubmodule(WarePath, SubPath));
+  Dialog.exec();
```
Comment (jctophefabre) [inline src/openfluid/ui/waresdev/WareSrcExplorer.cpp:469]: This seems to broke SOLID principles, but this code is maybe inspired by other cases in the wares/frament management? to be reviewed

**Comment by Arthoni** [comment]
Ready for review, last commit on generalization of regex can be left outside of pull if not relevant here.

**Review by jctophefabre — APPROVED** [review]

**Pull request merged**